app/services/ClaimToAdjustService.scala

```diff
@@ -48,24 +49,25 @@ class ClaimToAdjustService @Inject()(val financialDetailsConnector: FinancialDet
   private def getPoAPayments(documentDetails: List[DocumentDetail]): Either[Throwable, Option[TaxYear]] = {
     {
       for {
-        poa1 <- documentDetails.filter(_.documentDescription.exists(_.equals("ITSA- POA 1")))
-          .sortBy(_.taxYear).reverse.headOption.map(doc => makeTaxYearWithEndYear(doc.taxYear))
-        poa2 <- documentDetails.filter(_.documentDescription.exists(_.equals("ITSA - POA 2")))
-          .sortBy(_.taxYear).reverse.headOption.map(doc => makeTaxYearWithEndYear(doc.taxYear))
+        poa1 <- Option(documentDetails.filter(_.documentDescription.exists(_.equals("ITSA- POA 1")))
```

Comment:
we don't need Option wrapper here;
the whole reason we use for-comprehension is to extract poa1 and poa2

Reply:
+ I thin filter should be by "ITSA - POA 1", you missed one space after "ITSA"

Reply:
API spec says that ITSA- POA 1 is correct
https://confluence.tools.tax.service.gov.uk/pages/viewpage.action?pageId=367723014

Reply:
If we do not use the option wrapper here then the method will return Left(Exception) when the user has no PoA data, for example if they have just signed up to MTD. This means we would only be able to use it after the entry page - which is fine so long as people are aware of this

Reply:
this is not correct approach, we extract value from Option and wrap it back, why ?; we are in control what we return in each case, so we can always change this part of the logic